libraries/SITL/SIM_Precland.cpp

```diff
@@ -112,24 +119,30 @@ void SIM_Precland::update(const Location &loc, const Vector3d &position)
         return;
     }
 
-    const float distance_z = -position.z;
-    const float origin_height_m = _origin_height * 0.01f;
-    if (distance_z > _alt_limit + origin_height_m) {
-        _healthy = false;
-        return;
-    }
-
     const Location origin_center(static_cast<int32_t>(_origin_lat * 1.0e7f),
             static_cast<int32_t>(_origin_lon * 1.0e7f),
             static_cast<int32_t>(_origin_height),
             Location::AltFrame::ABOVE_HOME);
-    Vector2f center;
-    if (!origin_center.get_vector_xy_from_origin_NE(center)) {
+    Vector3f center;
```

Comment:
Maybe make this `centerf` and construct a `Vector3d center` from it; this will allow you to do simple vector subtraction below.

Reply:
Done.